lua/telescope/_extensions/dap.lua

```diff
@@ -231,35 +231,39 @@ local frames = function(opts)
 end
 
 return telescope.register_extension {
-  setup = function()
+  setup = function(ext_config)
     vim.cmd [[ highlight default link NvimDapSubtleFrame Comment ]]
 
-    require('dap.ui').pick_one = function(items, prompt, label_fn, cb)
-      local opts = {}
-      pickers.new(opts, {
-        prompt_title = prompt,
-        finder    = finders.new_table {
-          results = items,
-          entry_maker = function(entry)
-            return {
-              value = entry,
-              display = label_fn(entry),
-              ordinal = label_fn(entry),
-            }
+    overwrite_pick_one = vim.F.if_nil(ext_config.overwrite_pick_one, false)
```

Comment:
also this is a global variable without a `local` in front of it.

just put the vim.F.if_nil check inside the if condition: `if vim.F.if_nil(ext_config.overwrite_pick_one, false) then`